### PR TITLE
Refactors a bunch of code, removes our life hook

### PR DESCRIPTION
**Pull request by Carbonhell**
:cl:
fix: stamina HUD should update instantly now.
fix: Removed some useless code in an expensive proc, mobs should lag less
/:cl:

Remember, adding shit in Life() and similar should be avoided where possible.
Also moved shit around to clean up some tg files aswell. Please codecheck it carefully, some stuff like the catban edits weren't tested since they require a database and i cba to set one up.

### CODE REVIEW

File: hippiestation/code/_onclick/hud/combo.dm [hunk 1]
```diff
@@ -23,3 +23,6 @@
 				var/spacing = 16
 				img.pixel_x = -16 - spacing * length(streak) / 2 + spacing * i
 				add_overlay(img)
+
+/obj/screen/combo/proc/reset_streak()
```
Comment (Carbonhell) [inline hippiestation/code/_onclick/hud/combo.dm:27]: Tbh this isnt really needed

**Comment by JohnGinnane** [comment]
did you find out why we have our own progressbar?

**Comment by Carbonhell** [comment]
We have a fade in/out effect which tg does not have

**Comment by Carbonhell** [comment]
Progress bar works
Monk seems to work correctly, didn't see anything weird at a first glance
Cats and cluwnes are untested, they require a database connection to apply a jobban so just codecheck it correctly please

**Review by yoyobatty — APPROVED** [review]
Looks good for the most part

**Review by JohnGinnane — APPROVED** [review]

**Pull request merged**